src/extract.rs

```diff
@@ -168,3 +168,38 @@ impl ToTokens for InstArguments {
         self.0.to_tokens(tokens)
     }
 }
+
+pub struct TestAttrs(pub(crate) Punctuated<NestedMeta, Token![,]>);
```

Comment:
```suggestion
#[derive(Default)]
pub struct TestAttrs(pub(crate) Punctuated<NestedMeta, Token![,]>);
```

This suggestion would be the starting point of "flattening" an `Option<Punctuated<_>>` down to a `Punctuated` (using `Punctuated::default()` for the empty / missing case), which could alleviate a bit the code (since I conceptually consider a lack of `#[test_attributes(...)]` to be equivalent to `#[test_attributes()]`).

Reply:
Nice idea, thank you!